apps/src/lib/client/tx.rs

```diff
@@ -59,15 +58,29 @@ const TX_UNBOND_WASM: &str = "tx_unbond.wasm";
 const TX_WITHDRAW_WASM: &str = "tx_withdraw.wasm";
 const VP_NFT: &str = "vp_nft.wasm";
 
+/// Data needed for broadcasting a tx and
+/// monitoring its progress on chain
+///
+/// Txs may be either a dry run or else
+/// they should be encrypted and included
+/// in a wrapper.
+pub enum TxBroadcastData {
+    DryRun(Tx),
+    Wrapper {
+        tx: Tx,
+        wrapper_hash: String,
+        decrypted_hash: Option<String>,
+    },
+}
+
 pub async fn submit_custom(ctx: Context, args: args::TxCustom) {
     let tx_code = ctx.read_wasm(args.code_path);
     let data = args.data_path.map(|data_path| {
         std::fs::read(data_path).expect("Expected a file at given data path")
     });
     let tx = Tx::new(tx_code, data);
-    let (ctx, tx, keypair) = sign_tx(ctx, tx, &args.tx, None).await;
-    let (ctx, initialized_accounts) =
-        process_tx(ctx, &args.tx, tx, &keypair).await;
+    let (ctx, tx) = sign_tx(ctx, tx, &args.tx, None).await;
+    let (ctx, initialized_accounts) = process_tx(ctx, &args.tx, tx).await;
```

Comment:
not having the pass the keypair around is a nice improvement! I wonder if we could make it even nicer on the caller side, because we're always calling both of these together in here with e.g.:
```rust
let (ctx, initialized_accounts) = sign_and_process_tx(ctx, tx, &args.tx, None).await;
```

Reply:
(not necessarily in this PR)

Reply:
That's true. We could definitely merge those into one call